cromwell_tools/tests/test_cromwell_tools.py

```diff
@@ -45,68 +69,54 @@ def setUp(self):
         self.inputs_file2 = io.BytesIO(b"inputs_file2_content")
         self.options_file = io.BytesIO(b"options_file_content")
         self.label = io.BytesIO(b'{"test-label-key": "test-label-value"}')
-        self.url = "https://fake_url"
-        self.user = "fake_user"
-        self.password = "fake_password"
-        self.caas_key = "path/fake_key.json"
+
+    def _start_workflows(self, cromwell_auth, mock_request, _request_callback):
+        mock_request.post(cromwell_auth.url + '/api/workflows/v1', json=_request_callback)
+        return cromwell_tools.start_workflow(
+            self.wdl_file, self.inputs_file, cromwell_auth, self.options_file,
+            self.inputs_file2, self.zip_file, label=self.label)
 
     @requests_mock.mock()
-    def test_start_workflow(self, mock_request):
+    @mock.patch('cromwell_tools.cromwell_tools.generate_auth_header_from_key_file')
+    def test_start_workflow(self, mock_request, mock_header):
         """Unit test using mocks
         """
+        mock_header.return_value = {"Authorization": "bearer fake_token"}
+
         def _request_callback(request, context):
             context.status_code = 200
             context.headers['test'] = 'header'
             return {'request': {'body': "content"}}
 
-        # Check request actions
-        mock_request.post(self.url, json=_request_callback)
-        result = cromwell_tools.start_workflow(
-            self.wdl_file, self.inputs_file, self.url, self.options_file, self.inputs_file2, self.zip_file, self.user,
-            self.password, label=self.label)
-        self.assertEqual(result.status_code, 200)
-        self.assertEqual(result.headers.get('test'), 'header')
+        for cromwell_auth in auth_options:
```

Comment:
Instead of testing all of the auth options in one test, I would prefer splitting this into three different tests. It's slightly more repetitive but it makes it easier to tell what is failing when each test is testing one thing.

Reply:
I think one test with `pytest.mark.parametrize()` over the input types is the way to go. It means if you add another one in the future, all you need to do is add another input. I don't like rewriting test code.

Reply:
I like that idea! I keep forgetting about `pytest.mark.parametrize()`